### PR TITLE
Add support for Debug output in Analyzer

### PR DESCRIPTION
**Pull request by tlakollo**
When debugging applications via de console is very useful to have a Console print about the different values being used. In the case of debugging via the Test Explorer in Visual Studio, the Console output is not available to the user.

This PR adds the functionality to be able to choose to print to the Debug console that is available in Visual Studio while running a test via the Test Explorer
Adds enum to choose between Console and Debug output

### CODE REVIEW

File: src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimDataFlowAnalysis.cs [hunk 1]
```diff
@@ -80,20 +89,40 @@ public override void TraceVisitBlock (BlockProxy block)
 			if (!trace)
 				return;
 
-			Console.Write ("block " + block.Block.Ordinal + ": ");
+			TraceWrite ("block " + block.Block.Ordinal + ": ", tracingMechanism);
```
Comment (sbomer) [inline src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimDataFlowAnalysis.cs:92]: I think we might as well remove the `tracingMechanism` parameter and just check the field inside `TraceWriteLine`.

**Comment by sbomer** [comment]
I checked and it seems like the Console output is visible for failed testcases on macOS (when running with `dotnet test`), but not windows. :( I wonder if there's a way to make this more consistent.

File: src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimDataFlowAnalysis.cs [hunk 2]
```diff
@@ -80,20 +89,40 @@ public override void TraceVisitBlock (BlockProxy block)
...
+				Console.WriteLine (tracingInfo);
+			else if (tracingMechanism == TracingType.Debug)
+				Debug.WriteLine (tracingInfo);
+			else
+				throw new NotImplementedException (message: "invalid TracingType is being used");
+		}
+
+		private static void TraceWrite (string tracingInfo)
+		{
+			if (tracingMechanism == TracingType.Console)
```
Comment (sbomer) [inline src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimDataFlowAnalysis.cs:120]: nit: might be more appropriate as a switch

**Review by sbomer — APPROVED** [review]
Thanks!

**Comment by tlakollo** [comment]
> I checked and it seems like the Console output is visible for failed test cases on macOS (when running with `dotnet test`), but not windows. :( I wonder if there's a way to make this more consistent.

I found several issues open because of VS behavior and the inability to see the console output in tests. Apparently, NUnit might show a link with the console output in the test explorer but Xunit will not. Most of the provided answers are just "use Debug.Write/Debug.WriteLine"

**Pull request merged**